src/sparseml/pytorch/torch_to_onnx_exporter.py

```diff
@@ -0,0 +1,288 @@
+import warnings
+from copy import deepcopy
+from typing import Any, Dict, Iterable, List
+
+import onnx
+import torch
+from packaging import version
+
+from sparseml.exporters import transforms as sparseml_transforms
+from sparseml.exporters.base_exporter import BaseExporter
+from sparseml.exporters.transforms.base_transform import BaseTransform
+from sparseml.pytorch import _PARSED_TORCH_VERSION
+from sparseml.pytorch.opset import TORCH_DEFAULT_ONNX_OPSET
+from sparseml.pytorch.utils.helpers import tensors_module_forward, tensors_to_device
+from sparseml.pytorch.utils.model import is_parallel_model
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TorchToONNX(BaseExporter):
+    """
+    Transforms a `torch.nn.Module` into an `onnx.ModelProto` using `torch.onnx.export`.
+
+    Example usage:
+
+    ```python
+    model = torchvision.models.resnet18().eval()
+    exporter = TorchToONNX(sample_batch=torch.randn(1, 3, 224, 224))
+    exporter.export(model, "resnest18.onnx")
+    ```
+
+    :param sample_batch: the batch to export an onnx for, handles creating the
+        static graph for onnx as well as setting dimensions
+    :param opset: onnx opset to use for exported model.
+        Default is based on torch version.
+    :param disable_bn_fusing: torch >= 1.7.0 only. Set True to disable batch norm
+        fusing during torch export. Default and suggested setting is True. Batch
+        norm fusing will change the exported parameter names as well as affect
+        sensitivity analyses of the exported graph.  Additionally, the DeepSparse
+        inference engine, and other engines, perform batch norm fusing at model
+        compilation.
+    :param export_kwargs: kwargs to be passed as is to the torch.onnx.export api
+        call. Useful to pass in dyanmic_axes, input_names, output_names, etc.
+        See more on the torch.onnx.export api spec in the PyTorch docs:
+        https://pytorch.org/docs/stable/onnx.html
+    """
+
+    def __init__(
+        self,
+        sample_batch: Any,
+        opset: int = TORCH_DEFAULT_ONNX_OPSET,
+        disable_bn_fusing: bool = True,
+        **export_kwargs,
+    ):
+        super().__init__(
+            [
+                _TorchOnnxExport(
+                    sample_batch, opset, disable_bn_fusing, **export_kwargs
+                ),
+                sparseml_transforms.FoldIdentityInitializers(),
+                sparseml_transforms.FlattenQParams(),
+                sparseml_transforms.UnwrapBatchNorms(),
+                sparseml_transforms.DeleteTrivialOnnxAdds(),
+            ],
+        )
+
+    def pre_validate(self, module: Any) -> torch.nn.Module:
+        if not isinstance(module, torch.nn.Module):
+            raise TypeError(f"Expected torch.nn.Module, found {type(module)}")
+
+        if is_parallel_model(module):
+            module = module.module
+
+        return deepcopy(module).to("cpu").eval()
+
+    def post_validate(self, model: Any) -> onnx.ModelProto:
```

Comment:
```suggestion
    def post_validate(self, model:  torch.nn.Module) -> onnx.ModelProto:
```

Reply:
I think this should be onnx.ModelProto - I added that